sqlserver/datadog_checks/sqlserver/statements.py

```diff
@@ -447,8 +451,8 @@ def _collect_plans(self, rows, cursor, deadline):
             # for stored procedures, we only want to look up plans for the entire procedure
             # not every query that is executed within the proc. In order to accomplish this,
             # we use the procedure_signature as the plan key
-            if row['is_proc']:
-                plan_key = row['procedure_signature']
+            if row['is_proc'] or row['is_encrypted']:
+                plan_key = row['plan_handle'][0:16]
```

Comment:
Why are you taking the substring here?

Reply:
Ah ok! In that case, I would definitely advocate for a comment on optimizations like this.

By the way, TIL: I initially questioned if it would matter because the key is always hashed. So I started checking the size of dicts and was intrigued to find the same size of dict for all of the below conditions.

```python
>>> import sys
>>> d = dict()
>>> sys.getsizeof(d)
232
>>> d[1] = True
>>> sys.getsizeof(d)
232
>>> d['a'*999999] = True
>>> sys.getsizeof(d)
232
>>> d[5] = 'a'*999999
>>> sys.getsizeof(d)
232
```

It turns out that dicts allocate a [fixed size of dict for different unique key sizes](https://www.quora.com/Why-are-the-sizes-of-dictionaries-all-240-when-running-the-size-command-in-Python).

But then I realized that original keys are still stored because you can iterate over dict keys and get the original input values, so all of the above is moot.